tasks/internal/update-fbc-catalog-task/tests/test-update-fbc-catalog-default.yaml

```
@@ -0,0 +1,101 @@
+          value: "registry.io/image0@sha256:0000"
+        - name: fromIndex
+          value: "quay.io/scoheb/fbc-index-testing:latest"
+        - name: targetIndex
+          value: "quay.io/fbc/catalog:test"
+        - name: buildTags
+          value: "[]"
+        - name: addArches
+          value: "[]"
+        - name: iibServiceAccountSecret
+          value: "iib-service-account-secret"
+        - name: subdirectory
+          value: $(context.pipelineRun.uid)
+    - name: check-result
+      params:
+        - name: jsonBuildInfo
+          value: $(tasks.run-task.results.jsonBuildInfo)
+        - name: buildState
+          value: $(tasks.run-task.results.buildState)
+        - name: genericResult
+          value: $(tasks.run-task.results.genericResult)
+        - name: indexImageDigests 
+          value: $(tasks.run-task.results.indexImageDigests)
+        - name: iibLog
+          value: $(tasks.run-task.results.iibLog)
+        - name: exitCode
+          value: $(tasks.run-task.results.exitCode)
+      taskSpec:
+        params:
+          - name: jsonBuildInfo
+            type: string
+          - name: buildState
+            type: string
+          - name: genericResult
+            type: string
+          - name: indexImageDigests
+            type: string
+          - name: iibLog
+            type: string
+          - name: exitCode
+            type: string
+        steps:
+          - name: check-result
+            image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+            script: |
+              #!/bin/bash
+              set -x
+
+              # the jsonBuild mockes has 9 keys on it
+              keyLength=$(jq '. | length' <<< '$(params.jsonBuildInfo)')
+              if [ "$keyLength" -ne 9 ]; then
+                echo "The task did not save a valid json in jsonBuildInfo result"
+                exit 1
+              fi
+
+              # shellcheck disable=SC2046
```

what is this shellcheck warning wanting you to do?

---

it is for the "Quote this to prevent word splitting." error. Normally adding the quotes would just fix the warning, but for some reason using it in the tekton parameters with a json content, it adds some extra single quotes that breaks jq. I tried some other ways to not require the double quotes, but they were always causing another error/warning, so ignoring the error looked better. I improved the comment.

---

I don't think this is the right solution. There are a couple of issues. The reason why adding `""` breaks it is because the param has a weird value to begin with. Compare it to `jsonBuildInfo` which is a proper json. `buildState` has everything escaped, not sure why - it shouldn't be that way:
```
  Params:
    Name:   jsonBuildInfo
    Value:  {"id":1,"distribution_scope":"stage","fbc_fragment":"registry.io/image0@sha256:0000","internal_index_image_copy":"registry-proxy-stage.engineering.redhat.com/rh-osbs-stage/iib:1","logs":{"url":"https://fakeiib.host/api/v1/builds/1/logs"},"request_type":"fbc-operations","state":"failed","state_reason":"IIB Mocked Error","user":"iib@kerberos"}

    Name:   buildState
    Value:  "{\"state\":\"failed\",\"state_reason\":\"IIB Mocked Error\"}"
```
In the task, this should be changed to:
```
            jq -c '{ "state": .state, "state_reason": .state_reason }' "$(results.jsonBuildInfo.path)"  \
            | tee "$(results.buildState.path)"
```

But fixing that alone won't help. You have to realize how the param expansion works. Before the script is even started, `$(params. buildState)` will get substituted for its value. That's gonna be an issue if there are `"` in the string, because then the script will run with something like `<<< "{"key": "value"}"`. So because of this it is recommended to use env vars for params in Tekton. With that, it works:
```
            env:
              - name: BUILD_STATE
                value: "$(params.buildState)"
```
And in the script:
```
state="$(jq -cr '.state' <<< "$BUILD_STATE")"
```

---

Got it. I think you can get around that issue by using the `env` directive like https://github.com/konflux-ci/release-service-catalog/blob/9666883410181fdad34e660b31ae921fe58c499b/tasks/managed/reduce-snapshot/reduce-snapshot.yaml#L36 but I am fine with just disabling shellcheck here

---

@mmalina I will add the changes in the following up PR that will contain fixes/test for this task. Are you ok with that?

---

Sounds good.